GitUI/UserControls/RevisionGrid.cs

```diff
@@ -427,6 +428,78 @@ private void RevisionsKeyDown(object sender, KeyEventArgs e)
             }
         }
 
+        private void RevisionsKeyUp(object sender, KeyEventArgs e)
+        {
+            if (Revisions.RowCount < LastRowIndex || LastRowIndex < 0 || Revisions.RowCount == 0)
```

Comment:
I posted a [PR](https://github.com/RussKie/gitextensions/pull/1) to give you an idea of how I would see it.